test/e2e/common/security_context.go

```diff
@@ -50,44 +53,92 @@ var _ = framework.KubeDescribe("Security Context", func() {
-		  Description: Container created with runAsUser option, passing an id (id:65534) uses that
-		  given id when running the container.
-		  This test is marked LinuxOnly since Windows does not support running as UID / GID.
+			Release : v1.12
+			Testname: Security Context: runAsUser (id:65534)
+			Description: Container created with runAsNonRoot and runAsUser option, passing an id (id:65534) uses that given id when running the container and runAsNonRoot ensures that container does not ran as root.
+			This test is marked LinuxOnly since Windows does not support running as UID / GID.
 		*/
 		It("should run the container with uid 65534 [LinuxOnly] [NodeConformance]", func() {
-			createAndWaitUserPod(65534)
+			createAndWaitUserPod(true, 65534)
 		})
 
 		/*
-		  Release : v1.12
-		  Testname: Security Context: runAsUser (id:0)
-		  Description: Container created with runAsUser option, passing an id (id:0) uses that
-		  given id when running the container.
-		  This test is marked LinuxOnly since Windows does not support running as UID / GID.
+			Release : v1.12
+			Testname: Security Context: runAsUser (id:0)
+			Description: Container created with runAsUser option, passing an id (id:0) uses that
+			given id when running the container.
+			This test is marked LinuxOnly since Windows does not support running as UID / GID.
+			This test can not be promoted to [Conformance] because a platform may not allow containers to run with uid 0 and still be conformant.
 		*/
 		It("should run the container with uid 0 [LinuxOnly] [NodeConformance]", func() {
-			createAndWaitUserPod(0)
+			createAndWaitUserPod(false, 0)
+		})
+
+		/*
+			Release : v1.14
+			Testname: Security Context, runAsUser (non-root), runAsNonRoot
+			Description: Container is created with runAsNonRoot securityContext option without specifying default user to run container. Pod would succeed only if container run as non-root user.
+			A platform may not allow to run container as root and still be conformant.
+			This test is marked LinuxOnly since Windows does not support running as UID / GID.
+		*/
+		It("should not allow to run the container as root when RunAsNonRoot securityContext is set [LinuxOnly] [NodeConformance]", func() {
+			podName := fmt.Sprintf("non-root-busybox-%s", uuid.NewUUID())
+			pod := makeUserPod(podName, framework.BusyBoxImage, []string{"/bin/true"}, true, 0)
+
+			By("Creating Pod")
+			podClient.Create(pod)
+
+			err := wait.Poll(5*time.Second, 2*time.Minute, func() (bool, error) {
+				kubeletPod, err := f.PodClient().Get(podName, metav1.GetOptions{})
+				Expect(err).NotTo(HaveOccurred(), "Pod must exists")
+				for _, status := range kubeletPod.Status.ContainerStatuses {
+					if status.Name == podName {
+						switch {
+						case status.State.Waiting != nil:
+							if status.State.Waiting.Reason == "CreateContainerConfigError" &&
+								strings.Contains(status.State.Waiting.Message, "runAsUser breaks non-root policy") {
```

Comment:
[conformance-test-requirements](https://github.com/kubernetes/community/blob/20d3fe4cd54d65f74a13b37d2648ffefa067a4c9/contributors/devel/sig-architecture/conformance-tests.md#conformance-test-requirements) state:
>Examples of tests which are not eligible for promotion to conformance:
>- anything that checks optional Condition fields, such as Reason or Message, as these may change over time (however it is reasonable to verify these fields exist or are non-empty)

Reply:
Though it is using optional option `RunAsNonRoot ` from podSpec but very important for a Conformant platform which may not allow a root user to run container.

As this e2e is meant to go for Conformance, I can update code to check non-empty reason and message. (**Disadvantage:** no accuracy of e2e as covering broader reason of Container's waiting state)

_Other suggestions are highly appreciated._

Reply:
@bgrant0607 - here is an example of an ask, but given that the tests are only valid for their specific release are there use cases that actually exist where 

> anything that checks optional Condition fields, such as Reason or Message, as these may change over time (however it is reasonable to verify these fields exist or are non-empty) 

actually apply, and do we want to allow that?

Reply:
/assign @bgrant0607